Lib/multiprocessing/pool.py

```diff
@@ -437,7 +443,7 @@ def _handle_workers(pool):
         # is terminated.
         while thread._state == RUN or (pool._cache and thread._state != TERMINATE):
             pool._maintain_pool()
-            time.sleep(0.1)
+            pool._wait_for_updates(timeout=0.2)
```

Comment:
The timeout is needed for changes in `thread._state`

Reply:
asyncio uses the "self-pipe" pattern to wake up itself when it gets an event from a different thread or when it gets a Unix signal. Would it be possible to use a self-pipe (or something else) to wake up the wait when thread._state changes?

Reply:
We would need to manage also the case when `pool._cache` is empty.

Reply:
> @vstinner I have used the self-pipe pattern to receive notifications on pool._cache and thread._state changes. I maintained the 0.2 timeout for making sure the old behaviour is maintained if some change is not notified using the self._change_notifier queue (by mistake or because of external reasons).

Why 0.2 and not 0.1 or 1.0? I understand that replacing 0.1 with 0.2 doubles the latency of thread pool. Am I right?

Reply:
I used `0.2` to maintain backwards compatibility with the old behaviour. See my other comment explaining why I maintained the timeout.